packages/opentelemetry-exporter-collector/src/CollectorExporter.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportResult } from '@opentelemetry/base';
+import { NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { Attributes, Logger } from '@opentelemetry/types';
+
+import * as collectorTypes from './types';
+import { toCollectorSpan } from './transform';
+
+import { onInit, onShutdown, sendSpans } from './platform/index';
+
+/**
+ * Collector Exporter Config
+ */
+export interface CollectorExporterConfig {
+  hostName?: string;
+  logger?: Logger;
+  serviceName?: string;
+  spanAttributes?: Attributes;
+  url?: string;
+}
+
+const defaultServiceName = 'collector-exporter';
+const defaultCollectorUrl = 'http://localhost:55678/v1/trace';
+
+/**
+ * Collector Exporter
+ */
+export class CollectorExporter implements SpanExporter {
+  readonly serviceName: string;
+  readonly url: string;
+  readonly logger: Logger;
+  readonly hostName: string | undefined;
+  private _isShutdown: boolean = false;
+
+  /**
+   * @param config
+   */
+  constructor(config: CollectorExporterConfig = {}) {
+    this.serviceName = config.serviceName || defaultServiceName;
+    this.url = config.url || defaultCollectorUrl;
+    if (typeof config.hostName === 'string') {
+      this.hostName = config.hostName;
+    }
+
+    this.logger = config.logger || new NoopLogger();
+
+    this.shutdown = this.shutdown.bind(this);
+
+    // platform dependent
+    onInit(this.shutdown);
+  }
+
+  /**
+   * Export spans.
+   * @param spans
+   * @param resultCallback
+   */
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ) {
+    this._exportSpans(spans)
+      .then(() => {
+        resultCallback(ExportResult.SUCCESS);
+      })
+      .catch((status: number = 0) => {
+        if (status < 500) {
```

Comment:
We already have similar logic in zipkin too. We can argue which code can be retrayable, but basically, any 4xx code means that server is responding that you won't get what you want until you change something, that's why it is treated this way. Whereas 5xx means usually that server encountered some fatal unexpected error so this way without changing anything you might want to retry it and it might work